File: Medium-Array/LongestSubsequnence.java
// WAP to find the longest Subsequence

import java.util.HashSet;
import java.util.Scanner;

class LongestSubsequnence
{
    public static int longestConsecutive(int[] arr) 
    {
        HashSet<Integer> numSet = new HashSet<>();
        int longest = 0;

        // Add all numbers to the set
        for (int num : arr) {
            numSet.add(num);
        }

        // Iterate through each number
        for (int num : numSet) {
            // Check if it's the start of a sequence
            if (!numSet.contains(num - 1)) {
                int currentNum = num;
                int streak = 1;

                while (numSet.contains(currentNum + 1)) {
                    currentNum++;
                    streak++;
                }

                longest = Math.max(longest, streak);
            }
        }

        return longest;     
    }
    public static void main(String[] args)
    {
        Scanner sc=new Scanner(System.in);
        // Take array size as input
        System.out.println("Enter the size of array:");
        int n = sc.nextInt();
        int arr[] = new int[n];
        // Take array elements as input
        System.out.println("Enter the elements of array:");
        for(int i = 0; i < n; i++)
        {
            arr[i] = sc.nextInt();
        }
    }
}

File: Medium-Array/LongestSubsequnence.py
# WAP to find the longest Subsequence

def longestConsecutive(arr):
        n = len(arr)
        # If the array is empty
        if n == 0:
            return 0 

        # Initialize the longest sequence length
        longest = 1 
        st = set()

        # Put all the array elements into the set
        for i in range(n):
            st.add(arr[i])

        # Traverse the set to find the longest sequence
        for it in st:
            # Check if 'it' is a starting number of a sequence
            if it - 1 not in st:
                # Initialize the count of the current sequence
                cnt = 1 
                # Starting element of the sequence
                x = it 

                # Find consecutive numbers in the set
                while x + 1 in st:
                    # Move to the next element in the sequence
                    x = x + 1 
                    # Increment the count of the sequence
                    cnt = cnt + 1 
                # Update the longest sequence length
                longest = max(longest, cnt)
        return longest

arr = []
n = int(input("Enter the number of elements in the array: "))
for i in range(n):
    element = int(input("Enter element: "))
    arr.append(element)
longest = longestConsecutive(arr)
print("The length of the longest increasing subsequence is: ", longest)